src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -100,7 +102,9 @@ class UserService(protected val userInfo: UserInfo, dataSource: DataSource, prot
     // retrying this call will retry the failures, failures due to already created groups/entries are ok
     handleFutures(Future.sequence(Seq(
       toFutureTry(gcsDAO.createProxyGroup(user)),
-      toFutureTry(Future(dataSource.inTransaction()(txn => containerDAO.authDAO.saveUser(user, txn)))),
+      toFutureTry(dataSource.inFutureTransaction() { txn =>
+        Future(containerDAO.authDAO.saveUser(user, txn)).
+          flatMap(user => addUsersToAllUsersGroup(Seq(user), txn)) }),
```

Comment:
What's the difference between  
`toFutureTry(Future(dataSource.inTransaction() { blah` and  
`toFutureTry(dataSource.inFutureTransaction(){ Future( blah`?

(i.e. what does the combination of inFutureTransaction and pushing the Future into the txn achieve)